calico/etcdutils.py

```diff
@@ -310,5 +325,41 @@ def _on_snapshot_loaded(self, etcd_snapshot_response):
         pass
 
 
+def delete_empty_parents(client, child_key, root_key, timeout=DEFAULT_TIMEOUT):
+    """
+    Attempts to delete child_key and any empty parent directories.
+
+    Makes a best effort.  If any of the deletes fail, gives up.  This
+    method is safe, even if another client is writing to the directory (the
+    delete will fail if the directory becomes non-empty).
+
```

Comment:
Confusing comment, I think.  child_key does have to be a dir too.

Writing a key in etcd atomically creates the parent directories so I think we're good on the other concern.